docker-compose.yml

```diff
@@ -1541,37 +1544,6 @@ services:
         /arrow/ci/scripts/java_build.sh /arrow /build &&
         /arrow/ci/scripts/java_test.sh /arrow /build"
 
-  debian-java-jni:
-    # Includes plasma test, jni for gandiva and orc, and C data interface.
-    # Usage:
-    #   docker-compose build debian-java
-    #   docker-compose build debian-java-jni
-    #   docker-compose run debian-java-jni
-    image: ${REPO}:${ARCH}-debian-9-java-${JDK}-maven-${MAVEN}-jni
-    build:
-      context: .
-      dockerfile: ci/docker/linux-apt-jni.dockerfile
-      cache_from:
-        - ${REPO}:${ARCH}-debian-9-java-${JDK}-maven-${MAVEN}-jni
-      args:
-        llvm: ${LLVM}
-    shm_size: *shm-size
-    environment:
-      <<: *ccache
-      ARROW_BUILD_TESTS: "OFF"
-      ARROW_S3: "OFF"
-      ARROW_SUBSTRAIT: "OFF"
```

Comment:
Can you reuse these environment variables in the java-manylinux build?

Reply:
We can use `ARROW_BUILD_TEST=OFF` and `ARROW_SUBSTRAIT=OFF` and they are already used.
(But I want to enable `ARROW_BUILD_TEST` in ARROW-17081 / #13911 to run a Dataset JNI test that uses `arrow_testing`.)

We can't use `ARROW_S3=OFF` because our JNI packages enable S3 recently: ARROW-16584 / #13157